src/gui/dialogs/story_viewer.cpp

```diff
@@ -319,63 +319,52 @@ void story_viewer::display_part(window& window)
 void story_viewer::draw_floating_image(window& window, floating_image_list::const_iterator image_iter, int this_part_index)
 {
 	const auto& images = current_part_->get_floating_images();
+	canvas& window_canvas = window.get_canvas(0);
 
 	// If the current part has changed or we're out of images to draw, exit the draw loop.
-	if((this_part_index != part_index_) || (image_iter == images.end())) {
-		timer_id_ = 0;
-		return;
-	}
-
-	const auto& floating_image = *image_iter;
-
-	std::ostringstream x_ss;
-	std::ostringstream y_ss;
-
-	// Floating images are scaled by the same factor as the background.
-	x_ss << "(trunc(fi_ref_x * base_scale_x) + base_origin.x";
-	y_ss << "(trunc(fi_ref_y * base_scale_y) + base_origin.y";
-
-	if(floating_image.centered()) {
-		x_ss << " - (image_original_width  / 2)";
-		y_ss << " - (image_original_height / 2)";
-	}
-
-	x_ss << " where fi_ref_x = " << floating_image.ref_x() << ")";
-	y_ss << " where fi_ref_y = " << floating_image.ref_y() << ")";
-
-	config cfg, image;
+	while((this_part_index == part_index_) && (image_iter != images.end())) {
+		const auto& floating_image = *image_iter;
+		++image_iter;
 
-	image["x"] = x_ss.str();
-	image["y"] = y_ss.str();
-	image["w"] = floating_image.autoscale() ? "(width)"  : "(image_width)";
-	image["h"] = floating_image.autoscale() ? "(height)" : "(image_height)";
-	image["name"] = floating_image.file();
+		std::ostringstream x_ss;
+		std::ostringstream y_ss;
 
-	// TODO: implement handling of the tiling options.
-	//image["resize_mode"] = "tile_centered"
+		// Floating images are scaled by the same factor as the background.
```

Comment:
Just before creating this PR, I noticed that this should say "Floating image's locations are scaled ...". Hence opening it as a draft for now.